### PR TITLE
Support negative comparison on str_compare in StrEndsWith rule

### PR DESCRIPTION
**Pull request by ChrisDBrown**
Follow up to https://github.com/rectorphp/rector-src/pull/339 to support negative matching on str_compare transformations as well as substr.

EDIT: Sorry for multiple PRs here, didn't notice this was needed until after my first one. This should fully resolve the issue!

### CODE REVIEW

File: rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc [hunk 1]
```diff
@@ -6,11 +6,12 @@ class SubstrNotPrefix
 {
     public function run()
     {
-        $isMatch = $needle !== substr($haystack, -strlen($needle));
+        $isNotMatch = substr($haystack, -strlen($needle)) !== $needle;
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc:9]: I think you can add another line for flip check like other fixtures, so no need another file:

```suggestion
        $isNotMatch = substr($haystack, -strlen($needle)) !== $needle;
        $isNotMatch = $needle !== substr($haystack, -strlen($needle));
```

File: rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc [hunk 2]
```diff
@@ -6,11 +6,12 @@ class SubstrNotPrefix
...
     }
 }
 
 ?>
+
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc]: new line is not needed:

```suggestion
```

File: rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc [hunk 3]
```diff
@@ -20,7 +21,7 @@ class SubstrNotPrefix
 {
     public function run()
     {
-        $isMatch = !str_ends_with($haystack, $needle);
+        $isNotMatch = !str_ends_with($haystack, $needle);
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc:25]: I think you can add another line for flip check like other fixtures, so no need another file:

```suggestion
        $isNotMatch = !str_ends_with($haystack, $needle);
        $isNotMatch = !str_ends_with($haystack, $needle);
```

File: rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix_other_way.php.inc [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?php
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix_other_way.php.inc]: this fixture can be removed.

File: rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc (hunk 1 above)
Comment (ChrisDBrown) [reply rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc:9]: The positive match on substr is using two separate files at the moment - `fixture.php.inc` and `substr_other_way.php.inc`. Should I move this to a single file at the same time?

Comment (samsonasik) [reply rules-tests/Php80/Rector/Identical/StrEndsWithRector/Fixture/substr_not_prefix.php.inc:9]: I think yes, for grouping case as both equals, just flipped compare. `substr_other_way.php.inc` content can be added to `fixture.php.inc`, then remove `substr_other_way.php.inc`.

**Comment by ChrisDBrown** [comment]
Thanks for the feedback @samsonasik - I've addressed those points now.

**Comment by TomasVotruba** [comment]
Thank you @ChrisDBrown and @samsonasik  :+1:

**Pull request merged**